Make sure we calculate tOffset in ms.

TroubleMakerBen opened the pull request.

jaelim-ms commented on MobileCenterCrashes/MobileCenterCrashes/Internals/Utils/MSErrorLogFormatter.m:
```diff
@@ -297,8 +297,8 @@ + (MSErrorReport *)errorReportFromLog:(MSAppleErrorLog *)errorLog {
   NSDate *appStartTime =
-  [NSDate dateWithTimeIntervalSince1970:([errorLog.toffset doubleValue] - [errorLog.appLaunchTOffset doubleValue])];
-  NSDate *appErrorTime = [NSDate dateWithTimeIntervalSince1970:[errorLog.toffset doubleValue]];
+  [NSDate dateWithTimeIntervalSince1970:(([errorLog.toffset doubleValue]/1000) - ([errorLog.appLaunchTOffset doubleValue])/1000)];
```
You can do it like this. 😜 
`A/1000 - B/1000` ==`(A - B) / 1000`
One less division.

clpolet commented on MobileCenter/MobileCenter/Internals/Channel/MSLogManagerDefault.m:
```diff
@@ -91,7 +91,7 @@ - (void)processLog:(id<MSLog>)log withPriority:(MSPriority)priority {
 
   // Set common log info.
-  log.toffset = [NSNumber numberWithInteger:[[NSDate date] timeIntervalSince1970]];
+  log.toffset = [NSNumber numberWithInteger:([[NSDate date] timeIntervalSince1970] * 1000)];
```
We should add a macro or helper for the whole ([[NSDate date] timeIntervalSince1970] * 1000) thing. Otherwise we'll fall into it at some other points ;)

TroubleMakerBen replied:
I like having a category here.

MehrdadMzfr commented on MobileCenter/MobileCenter/Internals/Model/MSAbstractLog.m:
```diff
@@ -28,8 +27,8 @@ - (NSMutableDictionary *)serializeToDictionary {
-    // Set the toffset relative to current time. The toffset needs to be up to date.
-    NSInteger now = [[NSDate date] timeIntervalSince1970];
+    // Set the toffset relative to current time. The toffset needs to be up to date.    
+    NSInteger now = ([[NSDate date] timeIntervalSince1970] * 1000);
```
is NSInteger big enough? shouldn't is be long instead?

TroubleMakerBen replied:
NSInteger Max is big enough as it's actually defined as:
`#define NSIntegerMax    LONG_MAX`

TroubleMakerBen commented:
I addressed the comments and consolidated the Utilitys a little bit. All “helper” classes are now called *util instead of *helper or *utils.

I'm thinking of moving `MSFileUtil` and `MSSenderUtil` into `MSUtil`, too, but wanted to hear about your thoughts, first. And it will be a separate PR anyway, this one's already pretty big.
